debugging/coreclr/outputManager.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+type outputCallback<T> = (result: T) => string;
+
+export interface OutputManager {
+    append(content: string): void;
+    appendLine(content: string): void;
+    performOperation<T>(startContent: string, operation: (outputManager: OutputManager) => Promise<T>, endContent?: string | outputCallback<T>, errorContent?: string | outputCallback<Error>): Promise<T>;
+}
+
+export class DefaultOutputManager implements OutputManager {
+    private skipFirstLine: boolean = false;
+    private isShown: boolean = false;
+
+    constructor(private readonly outputChannel: vscode.OutputChannel, private readonly level: number = 0) {
+    }
+
+    public append(content: string): void {
+        if (this.level) {
+            const split = content.split(/[\n\r]/g);
+            const generateContent =
+                (c: string) => {
+                    return this.skipFirstLine ? c : this.generatePrefix(c);
+                };
+
+            for (let i = 0; i < split.length; i++) {
+                if (i < split.length - 1) {
+                    this.outputChannel.appendLine(generateContent(split[i]));
+                    this.skipFirstLine = false;
+                } else if (split[i].length > 0) {
+                    this.outputChannel.append(generateContent(split[i]));
+                    this.skipFirstLine = true;
+                } else {
+                    this.skipFirstLine = false;
+                }
+            }
+        } else {
+            this.outputChannel.append(content);
+        }
+    }
+
+    public appendLine(content: string): void {
+        if (this.level) {
+            this.outputChannel.append(this.generatePrefix());
+        }
+
+        this.outputChannel.appendLine(content);
+    }
+
+    public async performOperation<T>(startContent: string, operation: (outputManager: OutputManager) => Promise<T>, endContent?: string | outputCallback<T>, errorContent?: string | outputCallback<Error>): Promise<T> {
```

Comment:
This might be the right place to put in the callWithTelemetryAndErrorHandling call...?